contrib/win32/win32compat/shell-host.c

```diff
@@ -1124,16 +1125,60 @@ start_withno_pty(wchar_t *command)
 	GOTO_CLEANUP_ON_FALSE(SetHandleInformation(pipe_in, HANDLE_FLAG_INHERIT, 0));
 	GOTO_CLEANUP_ON_FALSE(SetHandleInformation(child_pipe_write, HANDLE_FLAG_INHERIT, 0));
 
-	/*TODO - pick this up from system32*/
-	cmd[0] = L'\0';
-	GOTO_CLEANUP_ON_ERR(wcscat_s(cmd, MAX_CMD_LEN, L"cmd.exe"));
+	/*
+	* check if the input needs to be processed (ex for CRLF translation)
+	* input stream needs to be processed when running the command
+	* within shell processor. This is needed when
+	*  - launching a interactive shell (-nopty)
+	*    ssh -T user@target
+	*  - launching cmd explicity
+	*    ssh user@target cmd
+	*  - executing a cmd command
+	*    ssh user@target dir
+	*  - executing a cmd command within a cmd
+	*    ssh user@target cmd /c dir
+	*/
+
+	if (!command)
+		process_input = TRUE;
+	else {
+		command_len = wcsnlen_s(command, MAX_CMD_LEN);
+		if ((command_len >= 3 && wcsncmp(command, L"cmd", 4) == 0) ||
```

Comment:
The input processing logic is exclusively for cmd.exe. Powershell is a shell processor/interpreter in itself. It should not be included here to give it independence to do whatever it wants with the input stream (ex. it could support direction keys, etc).